### PR TITLE
use healthz api in WaitForAPI

### PR DESCRIPTION
**Pull request by xilabao**
finish TODO

### CODE REVIEW

File: cmd/kubeadm/app/master/apiclient.go [hunk 1]
```diff
@@ -75,7 +76,12 @@ func CreateClientAndWaitForAPI(file string) (*clientset.Clientset, error) {
 func WaitForAPI(client *clientset.Clientset) {
 	start := time.Now()
 	wait.PollInfinite(kubeadmconstants.APICallRetryInterval, func() (bool, error) {
-		// TODO: use /healthz API instead of this
+		healthStatus := 0
```
Comment (dmmcquay) [inline cmd/kubeadm/app/master/apiclient.go]: It seems we want to do something like this instead of what using `ComponentStatuses`, not both of them.

**Review by dmmcquay — CHANGES_REQUESTED** [review]
I think that we don't want to have both the `/healthz` checks and the `ComponentStatuses` being checked. 

@luxas am I correct in my logic?

**Comment by luxas** [comment]
@jbeda @mikedanese @krousey What do you think of this in general?
Should we switch to querying `/healthz` or stick to ComponentStatuses?
This PR per se needs reworking but I'd like to know where we want to go here...

**Comment by mikedanese** [comment]
/healthz sounds more appropriate to me then componentstatus so I approve.

**Comment by krousey** [comment]
/healthz is the appropriate check for the API being available.

**Comment by dmmcquay** [comment]
@xilabao can you rebase and make the suggested changes?

**Comment by grodrigues3** [comment]
@k8s-bot ok to test

**Comment by xilabao** [comment]
@k8s-bot unit test this

**Comment by xilabao** [comment]
@dmmcquay PTAL

**Review by luxas — COMMENTED** [review]
We should check /healthz of the controller manager and scheduler as well, but good progress!

**Comment by mikedanese** [comment]
We shouldn't check /healthz on those components in WaitForAPI. In WaitForAPI, we should wait for only the API.

**Comment by mikedanese** [comment]
This change lgtm

**Review by dmmcquay — APPROVED** [review]

**Comment by dmmcquay** [comment]
@xilabao Please edit your PR comment

**Comment by xilabao** [comment]
@k8s-bot gce etcd3 e2e test this

**Comment by luxas** [comment]
We should still make sure the controller manager and scheduler comes up properly by checking their `/healthz`, but as @mikedanese pointed out, not in this specific function `WaitForAPI`

**Pull request merged**